From 4905cef92baa7c539f01197fc279e4b70602d4cc Mon Sep 17 00:00:00 2001
From: Orlando Vazquez <ovazquez@gmail.com>
Date: Tue, 15 Aug 2017 16:48:28 -0700
Subject: [PATCH] CNAPI-714 zones in nightly1 with cpu_caps set to 'null'
 causing test-servers test to fail Reviewed by: Marsell Kukuljevic
 <marsell@joyent.com> Reviewed by: Trent Mick <trentm@gmail.com> Approved by:
 Marsell Kukuljevic <marsell@joyent.com>

---
 test/test-servers.js | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/test/test-servers.js b/test/test-servers.js
index 05df839..2dd560e 100644
--- a/test/test-servers.js
+++ b/test/test-servers.js
@@ -362,13 +362,20 @@ function validateServer(t, server, options) {
 
             t.ok(UUID_RE.test(vm.owner_uuid), 'ensure vm.owner_uuid is a UUID');
 
-            var numAttr = ['max_physical_memory', 'quota', 'cpu_cap'];
+            var numAttr = ['max_physical_memory', 'quota'];
             numAttr.forEach(function (attr) {
                 if (typeof (vm[attr]) !== 'undefined') {
                     vmAttrTypeEqual(t, vm, attr, 'number');
                 }
             });
 
+            var optionalNumAttr = ['cpu_cap'];
+            optionalNumAttr.forEach(function (attr) {
+                if (vm[attr] !== undefined && vm[attr] !== null) {
+                    vmAttrTypeEqual(t, vm, attr, 'number');
+                }
+            });
+
             vmAttrTypeEqual(t, vm, 'last_modified', 'string');
             vmAttrTypeEqual(t, vm, 'state', 'string');
         }
-- 
2.21.0

